narwhal/primary/src/proposer.rs

```diff
@@ -202,8 +215,9 @@ impl Proposer {
 
         // Make a new header.
         let num_of_digests = self.digests.len().min(self.max_header_num_of_batches);
-        let header_digests: VecDeque<_> = self.digests.drain(..num_of_digests).collect();
-        let parents: Vec<_> = self.last_parents.drain(..).collect();
+        let header_digests = std::mem::take(&mut self.digests);
```

Comment:
It seems that we are no longer capping the number of digests included based on `num_of_digests` or am I getting this wrong? That is a precautionary measure to ensure that we don't overload the header it would be good to have it in place